### PR TITLE
Migrated Unit Tests & Linter from Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrated Unit Tests from Travis CI to Github Actions
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | N/A

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22567)

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -16,3 +16,89 @@ jobs:
...
+
+        -   uses: actions/checkout@v2
+            with:
+                fetch-depth: 0
+
+        -   name: PrestaShop Configuration
+            run: cp tests-legacy/parameters.yml.travis app/config/parameters.yml
+
+        -   name: Composer Install
+            run: composer install --ansi --prefer-dist --no-interaction --no-progress --quiet
```
Comment (kpodemski) [inline .github/workflows/php.yml:148]: shall we cache it? https://github.com/actions/cache/blob/main/examples.md#php---composer

File: .github/workflows/phpunit/parameters.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# This file is a used by travis install with specific setting.
```
Comment (matks) [inline .github/workflows/phpunit/parameters.yml]:
```suggestion
# This file is used by GitHub Actions
```
😉

**Comment by matks** [comment]
```
Run php install-dev/index_cli.php --language=en --country=fr --domain=localhost --db_server=127.0.0.1 --db_name=prestashop --db_create=1 --name=prestashop.unit.test --email=demo@prestashop.com --
***
PrestaShopException: Link to database cannot be established: SQLSTATE[HY000] [1045] Access denied for user 'root'@'172.17.0.1' (using password: NO) in /home/runner/work/PrestaShop/PrestaShop/classes/db/DbPDO.php:136
Stack trace:
```
Ideas:
- `mirromutth/mysql-action@v1.1` seems to use  root password `password` so install should use password
- `mirromutth/mysql-action@v1.1` is in a network? maybe something must be done to allow the install step to happen in a docker inside this network

File: .github/workflows/php.yml (hunk 1 above)
Comment (Progi1984) [reply .github/workflows/php.yml:148]: Good idea. 1m21 => 30s

**Review by NeOMakinG — DISMISSED** [review]
LGTM

File: .github/workflows/php.yml [hunk 3]
```diff
@@ -14,5 +14,164 @@ jobs:
...
+
+  phpstan:
+    name: PHP Static Analysis
+    runs-on: ubuntu-20.04
+    steps:
+        -   name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+                php-version: '7.4'
+                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
```
Comment (PierreRambaud) [inline .github/workflows/php.yml]:
```suggestion
                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv, simplexml
```

File: .github/workflows/php.yml [hunk 4]
```diff
@@ -14,5 +14,164 @@ jobs:
...
+    runs-on: ubuntu-latest
+    strategy:
+        matrix:
+            php: ['7.1', '7.2', '7.3', '7.4']
+    steps:
+        -   name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+                php-version: ${{ matrix.php }}
+                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
```
Comment (PierreRambaud) [inline .github/workflows/php.yml]:
```suggestion
                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv, simplexml
```

File: .github/workflows/php.yml [hunk 5]
```diff
@@ -14,5 +14,164 @@ jobs:
...
+      runs-on: ubuntu-latest
+      strategy:
+          matrix:
+              php: ['7.1', '7.2', '7.3', '7.4']
+      steps:
+        -   name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+                php-version: ${{ matrix.php }}
+                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
```
Comment (PierreRambaud) [inline .github/workflows/php.yml]:
```suggestion
                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv, simplexml
```

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+  yamllint:
+    name: YAML Lint
+    runs-on: ubuntu-latest
+    steps:
+      -   name: Setup PHP
+          uses: shivammathur/setup-php@v1
+          with:
+              php-version: '7.4'
+              extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
```
Comment (PierreRambaud) [inline .github/workflows/lint.yml]:
```suggestion
              extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv, simplexml
```

File: .github/workflows/lint.yml [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+  twiglint:
+      name: Twig Lint
+      runs-on: ubuntu-latest
+      steps:
+      -   name: Setup PHP
+          uses: shivammathur/setup-php@v1
+          with:
+              php-version: '7.4'
+              extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
```
Comment (PierreRambaud) [inline .github/workflows/lint.yml]:
```suggestion
              extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv, simplexml
```

**Review by jolelievre — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by matthieu-rolland** [comment]
thanx @Progi1984

**Pull request merged**